Packs/ZeroFox/Integrations/ZeroFox/ZeroFox.py

```diff
@@ -1017,86 +1024,180 @@ def get_exploits_content(
+        raise ValueError("Incorrect timestamp in last alert of fetch-incidents")
     max_update_time = (
         parsed_last_alert_timestamp + timedelta(milliseconds=1)
-    ).strftime(date_format)
-    next_run["last_fetched"] = max_update_time
-    next_run["last_offset"] = "0"
+    ).strftime(DATE_FORMAT)
+
+    def get_alert_ids(alert: dict[str, Any]) -> int:
+        return alert.get("id") or 0
+    processed_alerts_ids: list[int] = list(map(get_alert_ids, processed_alerts))
+
+    return incidents, next_offset, max_update_time, processed_alerts_ids
+
+
+def parse_last_fetched_date(
+    last_fetched_str: str | None,
+    first_fetch_time: str
+) -> datetime:
+    # If no last_fetched present, use default value
+    if not last_fetched_str:
+        last_fetched_str = first_fetch_time
+    last_fetched = parse_date(last_fetched_str, date_formats=(DATE_FORMAT,))
+    # If last_fetched is invalid, raise ValueError
+    if last_fetched is None:
+        raise ValueError("last_fetched param is invalid")
+    return last_fetched
+
+
+""" COMMANDS """
+
+
+def test_module(client: ZFClient) -> str:
+    """
+    Performs basic get request to get item samples
+    """
+    client.get_policy_types()
+    return "ok"
+
+
+def fetch_incidents(
+    client: ZFClient,
+    last_run: dict[str, str],
+    first_fetch_time: str
+) -> tuple[dict[str, Any], list[dict[str, Any]]]:
+    # Last fetched date
+    last_fetched_str = last_run.get("last_fetched", "")
+    last_fetched = parse_last_fetched_date(last_fetched_str, first_fetch_time)
+    last_fetched_str = last_fetched.strftime(DATE_FORMAT)
+
+    # Saved offset of last run
+    last_offset_str: str = last_run.get("last_offset", "0")
+    last_offset = int(last_offset_str)
+
+    # Date of first run
+    first_run_at_str = last_run.get("first_run_at", "")
+    first_run_at = parse_last_fetched_date(first_run_at_str, first_fetch_time)
+
+    # Last modified fetch date
+    last_modified_fetched_str = last_run.get("last_modified_fetched", "")
+    last_modified_fetched = parse_last_fetched_date(last_modified_fetched_str, first_fetch_time)
+    last_modified_fetched_str = last_modified_fetched.strftime(DATE_FORMAT)
+
+    # Saved modified alerts offset of last run
+    last_modified_offset_str: str = last_run.get("last_modified_offset", "0")
+    last_modified_offset = int(last_modified_offset_str)
+
+    # ZeroFox Alert IDs previously created
+    zf_ids: list[int] = []
+    stored_zf_ids = last_run.get("zf-ids")
+    if isinstance(stored_zf_ids, list):
+        zf_ids = list(stored_zf_ids)
+
+    next_run = {
+        "last_fetched": last_fetched_str,
+        "last_offset": last_offset_str,
+        "first_run_at": first_run_at.strftime(DATE_FORMAT),
+        "last_modified_fetched": last_modified_fetched_str,
+        "last_modified_offset": last_modified_offset_str,
+        "zf-ids": zf_ids,
+    }
+
+    # Fetch new alerts
+    params = {
+        "min_timestamp": last_fetched,
```

Comment:
```
        "min_timestamp": last_fetched_str,
```
You may want to send a str to the params (not a date object)

Reply:
This wasn't an issue because the client parsed it correctly. But I changed anyway.